src/lib/download.py

```diff
@@ -0,0 +1,69 @@
+"""Module providing artifact downloading"""
+import argparse
+import os
+from urllib.parse import urlparse
+from requests import get
+from requests.auth import HTTPBasicAuth
+
+def download(url, username, password, output_dir, output_file):
+    """
+    ## This function downloads a file from a URL.
+
+    ### Args:
+        url (string): URL to download the file from
+        username (string): Username for authentication
+        password (string): Password for authentication
+        OUTPUT_DIR (string): Directory to store the downloaded file
+        output_file (string): Name of the downloaded file
+    """
+    if os.environ.get("URL") is not None and os.environ.get("URL") != "None":
+        url = os.environ.get("URL")
+    if os.environ.get("USERNAME") is not None and os.environ.get("USERNAME") != "None":
+        username = os.environ.get("USERNAME")
+    if os.environ.get("PASSWORD") is not None and os.environ.get("PASSWORD") != "None":
+        password = os.environ.get("PASSWORD")
+    if os.environ.get("OUTPUT_DIR") is not None and os.environ.get("OUTPUT_DIR") != "None":
+        output_dir = os.environ.get("OUTPUT_DIR")
+    if os.environ.get("OUTPUT_FILE") is not None and os.environ.get("OUTPUT_FILE") != "None":
+        output_file = os.environ.get("OUTPUT_FILE")
```

Comment:
It's weird behavior, it means environment has priority over argument on CLI, normally it's the opposite